analysis_options.yaml

```diff
@@ -71,7 +71,6 @@ linter:
     - flutter_style_todos
     - hash_and_equals
     - implementation_imports
-    # - invariant_booleans # too many false positives: https://github.com/dart-lang/linter/issues/811
```

Comment:
Ha! False positive. Although you uncovered the bidirectional iterator silly @srawlins !

Reply:
BidirectionalIterator cleanup landed in https://github.com/google/quiver-dart/pull/700.

Now that it's deprecated, we can probably stop implementing it in `TreeIterator`. I'll send a patch and bump the major version since it'll be a breaking change.